File: backend/app/schemas/lesson.py
from pydantic import BaseModel
from datetime import datetime


class LessonCreate(BaseModel):
    title: str
    description: str
    category: str | None = None  # success, improvement, error
    project_phase: str | None = None
    recommendation: str | None = None


class LessonUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    category: str | None = None
    project_phase: str | None = None
    recommendation: str | None = None


class LessonResponse(BaseModel):
    id: int
    folio: str
    title: str
    description: str
    category: str | None
    project_phase: str | None
    recommendation: str | None
    project_id: int
    recorded_by_id: int | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/lesson.ts
/**
 * Lesson learned type definitions.
 * Aligned with backend schemas: LessonResponse.
 */

/** Matches backend LessonResponse schema */
export interface Lesson {
  id: number;
  folio: string;
  title: string;
  description: string | null;
  category: string | null;
  project_phase: string | null;
  recommendation: string | null;
  project_id: number;
  recorded_by_id: number | null;
  created_at: string;
}

/** Cross-project lesson view used in LessonsPage (with project info) */
export interface LessonWithProject {
  id: number;
  folio: string;
  title: string;
  description: string;
  category: string;
  project_phase: string;
  recommendation: string;
  project_name: string;
  project_id: number;
  recorded_by: string;
  created_at: string;
}
